hack/jenkins/e2e.sh

```diff
@@ -605,32 +612,32 @@ case ${JOB_NAME} in
     : ${FAIL_ON_GCP_RESOURCE_LEAK:="true"}
     ;;
 
-  # Runs non-flaky tests on GCE on the release candidate branch,
-  # sequentially. As a reminder, if you need to change the skip list
-  # or flaky test list on the release branch, you'll need to propose a
-  # pull request directly to the release branch itself.
-  kubernetes-e2e-gce-release-1.1)
-    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-release-1.1"}
+  # Runs non-flaky tests on GCE sequntially, against the latest current release
+  # ci.  As a reminder, if you need to change the skip list or flaky test list
```

Comment:
You can remove the bit about prosing a pull request directly to the release branch, it doesn't apply (and imo, shouldn't) to 1.1 jobs, which pull read this script from HEAD to avoid that hassle.